compiler/src/dmd/expressionsem.d

```diff
@@ -1155,6 +1155,43 @@ L1:
     return e1;
 }
 
+/*
+ * Check whether `outerFunc` and `calledFunc` have the same `this`.
+ * If `calledFunc` is the member of a base class of the class that contains
+ * `outerFunc` we consider that they have the same this.
+ *
+ * Returns:
+ *  `true` if outerFunc and calledFunc may use the same `this` pointer.
+ * `false` otherwise.
+ */
+private bool haveSameThis(FuncDeclaration outerFunc, FuncDeclaration calledFunc)
+{
+    auto thisAd = outerFunc.isMemberLocal();
+    if (!thisAd)
+        return false;
```

Comment:
It seems that this cant be covered because you call the new function pnly for members funcs, e.g line 7247.

```d
if (memberFunc && haveSameThis(memberFunc, exp.f))
```

so you could assert on `thisAd` instead

Reply:
Yes, I was aware of that, but I figured that this function could be used from other contexts so it should not make assumptions about the status at the call site.

Reply:
ok